app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -386,10 +386,13 @@ def remove_middleware_datasource(ems_ref)
     end
 
     def emit_middleware_notification(type, op_name, op_arg, ems_ref)
+      mw_deployment = MiddlewareDeployment.find_by_ems_ref(ems_ref)
+      mw_server = mw_deployment.nil? ? MiddlewareServer.find_by_ems_ref(ems_ref) :
```

Comment:
@bronaghs, @agrare, I don't really know the class name at this point. What I get is the `ems_ref` which is the identifier string we see (and want to simplify) in the notifications (eg: `/t;hawkular/f;bf007c00-e3c5-4f18-8784-8b084bd9327a/r;Local~~`).

What happens is that it can either be for a MW Server or a Deployment (or, in the future, a DataSource, Messaging, Domain, etc). So, this is what it looks for a 
- MW Server: `/t;hawkular/f;bf007c00-e3c5-4f18-8784-8b084bd9327a/r;Local~~`
- MW Deployment: `/t;hawkular/f;bf007c00-e3c5-4f18-8784-8b084bd9327a/r;Local~~/r;Local~%2Fdeployment%3Dwildfly-helloworld.war`
- MW Datasource: `/t;hawkular/f;bf007c00-e3c5-4f18-8784-8b084bd9327a/r;Local~~/r;Local~%2Fsubsystem%3Ddatasources%2Fdata-source%3DExampleDS`
- ...

I see two alternatives to what we have now:
a) Parse the ems_ref string to get the Feed and Server IDs out of it and use it directly in the notification;
b) Figure out from the ems_ref string which class I'll be looking up for, ie, if it has `~%2Fdeployment%3D` it's a `MiddlewareDeployment`, if it has `~%2Fsubsystem%3Ddatasources%2Fdata-source%3D` it's a `MiddlewareDatasource`, etc.. and if none of the above, fallback to a `MiddlewareServer`.

While a) is the most simple as it won't need any lookup (and if the operation failed due to somehow the resource not being present in the DB, it won't fail in the notification as well), I wanted to have the reference to the actual object as I wanted to pass it as a `$subject` to the notification so that in the future it can be hyperlinked, so in that sense b) is more future-ready. But if we want to simplify and then when the linking is in place we can change, we can go with a), WDYT ?